pycobertura/cobertura.py

```diff
@@ -368,6 +371,13 @@ def file_source(self, filename):
         # Build a dict of lineno2 -> lineno1
         lineno_map = reconcile_lines(lines2, lines1)
 
+        # when we are using the same coverage file for both sides, we need to
+        # translate the coverage of lines1 so that it corresponds to its
+        # real lines.
+        if self.cobertura1 == self.cobertura2:
+            rev_map = {v: k for k, v in lineno_map.items()}
+            line_statuses1 = {rev_map[v]: v for v in lineno_map.values()}
```

Comment:
I'm looking a little closer at this code, and wonder... It seems that `line_statuses1` doesn't hold the same value types as before. When I add `print()` statements like this:

```diff
diff --git a/pycobertura/cobertura.py b/pycobertura/cobertura.py
index 75315b7..24c26fb 100644
--- a/pycobertura/cobertura.py
+++ b/pycobertura/cobertura.py
@@ -375,8 +375,13 @@ class CoberturaDiff(object):
         # translate the coverage of lines1 so that it corresponds to its
         # real lines.
         if self.cobertura1 == self.cobertura2:
+            print()
+            print('line_statuses1 (pre):', line_statuses1)
+            print('lineno_map:', lineno_map)
             rev_map = {v: k for k, v in lineno_map.items()}
+            print('rev_map:   ', rev_map)
             line_statuses1 = {rev_map[v]: v for v in lineno_map.values()}
+            print('line_statuses1 (post):', line_statuses1)
 
         lines = []
         for lineno, source in enumerate(lines2, start=1):
```

... then the output of the new test looks like this:

```
$ pytest -sk test_diff_same_report_different_source_dirs

...

tests/test_cobertura_diff.py::test_diff_same_report_different_source_dirs 
line_statuses1 (pre): {1: True, 2: True, 3: False, 4: True, 6: True, 7: True, 8: False, 9: True, 11: True, 12: True}
lineno_map: {1: 1, 4: 2, 5: 3, 6: 4, 7: 5, 8: 6, 9: 7, 10: 8, 11: 9, 12: 10}
rev_map:    {1: 1, 2: 4, 3: 5, 4: 6, 5: 7, 6: 8, 7: 9, 8: 10, 9: 11, 10: 12}
line_statuses1 (post): {1: 1, 4: 2, 5: 3, 6: 4, 7: 5, 8: 6, 9: 7, 10: 8, 11: 9, 12: 10}
```

Normally, `line_statuses1` should contain the line number as the key and a boolean as the value to show whether the line has been covered, but here we have line numbers as keys and values. Can you elaborate on your thought process here?

Reply:
ugh, this must have gone missing in a refactor... let me fix that o_O

Reply:
Ah this has gotten lost in local refactorings I guess. It's odd that this would not break other tests. Now checking if I can cover this better (along with the pre-refactoring version --- which I like more for readability anyway).